### PR TITLE
Add integration configuration docs

### PR DESCRIPTION
**Pull request by bastienbeurier**

### CODE REVIEW

File: docs-v2/reference/integration-configuration.mdx [hunk 1]
```diff
@@ -4,6 +4,318 @@ sidebarTitle: 'Integration config (nango.yaml)'
...
+### Sync fields
+
+Sync configuration fields are under `integrations.<INTEGRATION-ID>.syncs.<SYNC-NAME>`.
+
+<ResponseField name="description" type="string" required>
+    Describes the sync.
+</ResponseField>
+
+<ResponseField name="output" type="string" required>
+    Defines the schema of the data you want to sync. References a schema from the [Models](#model-fields) section of this configuration file.
```
Comment (khaliqgant) [inline docs-v2/reference/integration-configuration.mdx:77]: Output can also be an array. 

Intuitively I still find it a bit weird that the output of a sync is an array of the model but here we say the output is just a model

**Review by bodinsamuel — APPROVED** [review]

File: docs-v2/reference/integration-configuration.mdx [hunk 2]
```diff
@@ -4,6 +4,318 @@ sidebarTitle: 'Integration config (nango.yaml)'
...
+    Defaults to no scope. 
+</ResponseField>
+
+<ResponseField name="auto_start" type="boolean">
+    If `true`, automatically starts synchronizing between the external API and Nango when a new connection is created. Otherwise, it needs to be triggered via the API or Nango UI.
+
+    Defaults to `true`. 
+</ResponseField>
+
+<ResponseField name="track_delete" type="boolean">
```
Comment (khaliqgant) [inline docs-v2/reference/integration-configuration.mdx]: track_deletes

https://github.com/NangoHQ/nango/blob/d5e3607bfce6874df87313dea2b0a83c3b25f40e/packages/cli/lib/templates/nango.yaml#L12

File: docs-v2/reference/integration-configuration.mdx [hunk 3]
```diff
@@ -4,6 +4,318 @@ sidebarTitle: 'Integration config (nango.yaml)'
...
+```
+
+### Dynamic field keys
+
+You can allow dynamic keys in models:
+```yaml nango.yaml
+models:
+  Task:
+    id: string
+    __string: string
```
Comment (khaliqgant) [inline docs-v2/reference/integration-configuration.mdx:287]: might be good to show what this translates to in a typescript model

File: docs-v2/reference/integration-configuration.mdx (hunk 1 above)
Comment (bastienbeurier) [reply docs-v2/reference/integration-configuration.mdx:77]: Yes, I intentionally omitted it for clarity (same for endpoints). But again today, a customer needed some model dependency that was best solved by a multi-model sync, so I agree it should be documented. I'll add it!

**Pull request merged**